embassy-imxrt/src/iopctl.rs

```diff
@@ -0,0 +1,715 @@
+    ///
+    /// If there is any doubt, it is best to enable the input buffer.
+    ///
+    /// See Section 7.4.2.3 of reference manual.
+    fn enable_input_buffer(&self) -> &Self;
+
+    /// Disables the input buffer of a pin.
+    fn disable_input_buffer(&self) -> &Self;
+
+    /// Sets the slew rate of a pin.
+    ///
+    /// This controls the speed at which a pin can toggle,
+    /// which is voltage and load dependent.
+    fn set_slew_rate(&self, slew_rate: SlewRate) -> &Self;
+
+    /// Sets the output drive strength of a pin.
+    ///
+    /// A drive strength of [`DriveStrength::Full`] has twice the
+    /// high and low drive capability of the [`DriveStrength::Normal`] setting.
+    fn set_drive_strength(&self, strength: DriveStrength) -> &Self;
+
+    /// Enables the analog multiplexer of a pin.
+    ///
+    /// This must be called to allow analog functionalities of a pin.
+    ///
+    /// To protect the analog input, [`IopctlPin::set_function`] should be
+    /// called with [`Function::F0`] to disable digital functions.
+    ///
+    /// Additionally, [`IopctlPin::disable_input_buffer`] and [`IopctlPin::set_pull`]
+    /// with [`Pull::None`] should be called.
+    fn enable_analog_multiplex(&self) -> &Self;
+
+    /// Disables the analog multiplexer of a pin.
+    fn disable_analog_multiplex(&self) -> &Self;
+
+    /// Sets the ouput drive mode of a pin.
+    ///
+    /// A pin configured as [`DriveMode::OpenDrain`] actually operates in
+    /// a "pseudo" open-drain mode which is somewhat different than true open-drain.
+    ///
+    /// See Section 7.4.2.7 of reference manual.
+    fn set_drive_mode(&self, mode: DriveMode) -> &Self;
+
+    /// Sets the input inverter of an input pin.
+    ///
+    /// Setting this to [`Inverter::Enabled`] will invert
+    /// the input signal.
+    fn set_input_inverter(&self, inverter: Inverter) -> &Self;
+
+    /// Returns a pin to its reset state.
+    fn reset(&self) -> &Self;
+}
+
+/// Represents a pin peripheral created at run-time from given port and pin numbers.
+pub struct AnyPin {
+    pin_port: u8,
+    reg: &'static PioM_N,
+}
+
+impl AnyPin {
+    /// Creates a pin from raw port and pin numbers which can then be configured.
+    ///
+    /// This should ONLY be called when there is no other choice
+    /// (e.g. from a type-erased GPIO pin).
+    ///
+    /// Otherwise, pin peripherals should be configured directly.
+    ///
+    /// # Safety
+    ///
+    /// The caller MUST ensure valid port and pin numbers are provided,
+    /// and that multiple instances of [`AnyPin`] with the same port
+    /// and pin combination are not being used simultaneously.
+    ///
+    /// Failure to uphold these requirements will result in undefined behavior.
+    ///
+    /// See Table 297 in reference manual for a list of valid
+    /// pin and port number combinations.
+    #[must_use]
+    pub unsafe fn new(port: u8, pin: u8) -> Self {
```

Comment:
In other hals we use `steal` for this purpose.

Reply:
I'll change, sure :)